packages/combo-box/src/vaadin-combo-box-scroller.js

```diff
@@ -259,7 +259,9 @@ export class ComboBoxScroller extends PolymerElement {
       this.requestContentUpdate();
     }
 
-    if (index >= 0) {
+    // Do not jump back to the previously focused item while loading
+    // when requesting next page from the data provider on scroll.
+    if (index >= 0 && !this.loading) {
```

Comment:
Just to make sure, will the scroll happen after everything is loaded?

Reply:
No. That's intentional, otherwise the focus would still jump back to the previously focused item.
This is not what the user would expect to happen after scrolling down. 

The root cause of the original issue is that `filteredItems` can change in 2 different ways:
1. As a result of scrolling that causes more items to be loaded from the `dataProvider`,
2. As a result of typing that causes `filter` to change and also triggers `_filteredItemsChanged`.

We don't need to scroll in the first case while scrolling, as that would cause it to jump.